master/internal/task/allocation.go

```diff
@@ -197,7 +197,7 @@ func (a *Allocation) Receive(ctx *actor.Context) error {
 					})
 					a.model.State = a.state
 					a.model.IsReady = a.logBasedReadinessPassed
-					err := a.db.UpdateAllocationState(&a.model)
+					err := a.db.UpdateAllocationState(a.model)
 					if err != nil {
```

Comment:
my preference, no pressure
```suggestion
					if err := a.db.UpdateAllocationState(a.model); err != nil {
```